CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.10)
```

Comment:
I am worry, that CMake available on Travis CI is 3.9.2

Reply:
> I am not CMake expert, so I cannot comment all changes, but it looks good overall. I am fine with the changes when it doesn't break my or CI builds :-) Can you look why they are failing? Github's "MSYS using gcc and cmake" is broken for long time (tool installation issue), you may ignore that one I believe.

I'm not finished yet. There are a lot of weird bugs in the CI systems. ;-)

Reply:
Ok, thanks :-) Please consider to split it to smaller isolated changes if possible...